autotest/utilities/test_gdalwarp_lib.py

```diff
@@ -4056,3 +4056,71 @@ def test_gdalwarp_lib_conflicting_source_metadata(tmp_vsimem):
     out_ds = gdal.Warp("", [src_ds1, src_ds2], options="-of MEM -cvmd conflicting")
     assert out_ds.GetMetadataItem("FOO") == "conflicting"
     assert out_ds.GetMetadataItem("BAR") == "BAZ"
+
+
+###############################################################################
+# Test issue GH #9467
+
+
+def test_target_extent_consistent_size():
+    """Test issue GH #9467 where the output size is not consistent when using target extent
+    with different input datasets having the same resolution and CRS but different extent."""
+
+    # Create a source dataset with CRS 32613
+    src_ds_1 = gdal.GetDriverByName("MEM").Create("", 10980, 10980)
+    src_ds_1.SetProjection("EPSG:32613")
+    src_ds_1.SetGeoTransform((300000.0, 10.0, 0.0, 3900000.0, 0.0, -10.0))
+
+    src_ds_2 = gdal.GetDriverByName("MEM").Create("", 10980, 10980)
+    src_ds_2.SetProjection("EPSG:32613")
+    src_ds_2.SetGeoTransform((300000.0, 10.0, 0.0, 4000020.0, 0.0, -10.0))
+
+    bbox = (
+        -106.40856573808874,
+        35.10139620198477,
+        -105.92962613828232,
+        35.51543260935861,
+    )
+
+    ds = gdal.Warp(
+        "",
+        [src_ds_1, src_ds_2],
+        options=gdal.WarpOptions(
+            multithread=True,
+            dstSRS="EPSG:4326",
+            outputBounds=bbox,
+            resampleAlg="lanczos",
+            dstNodata=0,
+            srcNodata=0,
+            format="MEM",
+        ),
+    )
+
+    assert ds.RasterXSize == 4793
```

Comment:
In addition to the checksum we should check the resolution returned by ds.GetGeoTransform(). Same for second part of the test